crd-generator/api/src/main/java/io/fabric8/crd/generator/CustomResourceInfo.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator;
+
+import io.fabric8.kubernetes.api.model.Namespaced;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.model.Scope;
+import io.fabric8.kubernetes.model.annotation.ShortNames;
+import io.sundr.codegen.functions.ClassTo;
+import io.sundr.codegen.model.TypeDef;
+import io.sundr.codegen.model.TypeRef;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class CustomResourceInfo {
+  private final String group;
+  private final String version;
+  private final String kind;
+  private final String singular;
+  private final String plural;
+  private final String[] shortNames;
+  private final boolean storage;
+  private final boolean served;
+  private final Scope scope;
+  private final Optional<TypeRef> status;
+  private final TypeDef definition;
+  private final String crClassName;
+  private final String specClassName;
+  private final String statusClassName;
+
+  public CustomResourceInfo(String group, String version, String kind, String singular,
+    String plural, String[] shortNames, boolean storage, boolean served,
+    Scope scope, Optional<TypeRef> status, TypeDef definition, String crClassName,
```

Comment:
Optional shouldn't be used as method/constructor argument (for many reasons, but mainly I can pass a null value as an argument and it would defeat the purpose).

If you want to store the variable as an Optional, the constructor should use the target Type and then instantiate the variable as `Optional.ofNullabel(status)`

Reply:
Done.